plugins/inputs/mesos/mesos.go

```diff
@@ -418,29 +430,62 @@ var client = &http.Client{
 	Timeout:   time.Duration(4 * time.Second),
 }
 
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.URLEncoding.EncodeToString([]byte(auth))
+}
+
+func getEndpointParams(address string) map[string]string {
```

Comment:
This function seems a bit unecessary. Basically all you are doing is taking apart the given URL into three different parts, and then you later just put it back together exactly as it was already provided.